### PR TITLE
AP-2303 Share Dialog

### PR DESCRIPTION
**Pull request by dreamind**

### CODE REVIEW

File: Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/util/AccessType.java [hunk 1]
```diff
@@ -27,34 +27,51 @@
...
+        return Arrays.asList(AccessType.values()).stream()
+                .filter(accessType -> accessType.label.equals(label))
+                .findFirst()
+                .orElse(AccessType.NONE);
+    }
+
+    public static AccessType getAccessType(boolean isRead, boolean isWrite, boolean isOwner) {
         return Arrays.asList(AccessType.values()).stream().filter(accessType -> accessType.isRead == isRead &&
                 accessType.isWrite == isWrite &&
                 accessType.isOwner == isOwner).findFirst().orElse(AccessType.NONE);
```
Comment (brucenguyen1) [inline Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/util/AccessType.java:77]: These parameters can be reduced and encapsulated as an AccessRights object.

**Review by brucenguyen1 — APPROVED** [review]

File: Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/common/access/Assignment.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+/**
+ * Assignment model for listmodel-based UI
+ */
+public class Assignment {
+    public String name;
+    public String rowGuid;
+    public Type type;
+    public boolean isGroup;
+    public String access; // Viewer, Editor, Owner (UI front for AccessType)
```
Comment (nolantellis) [inline Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/common/access/Assignment.java:36]: Use the existing Enum here. U can use Capitalize to get the Capital String.
Also U can get the Enum back from the String using AccessType.valueOf(String) instead of the method.

File: Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/util/AccessType.java [hunk 3]
```diff
@@ -27,34 +27,51 @@
...
+ * | Owner  | 1        | 1         | 1             |
  */
 @Getter
 public enum AccessType {
 
-    NONE(false, false, false),
-    VIEWER(true, false, false),
-    EDITOR(true, true, false),
-    OWNER(true, true, true);
+    NONE(false, false, false, "None"),
```
Comment (nolantellis) [inline Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/util/AccessType.java:40]: No need for those Labels.

File: Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/ShareController.java [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+        applyChanges();
+        getSelf().detach();
+    }
+
+    @Listen("onClick = #btnCancel")
+    public void onClickBtnCancel() {
+        getSelf().detach();
+    }
+
+}
```
Comment (nolantellis) [inline Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/ShareController.java:365]: This class needs Refactoring

File: Apromore-Database/src/main/java/org/apromore/dao/GroupRepository.java [hunk 5]
```diff
@@ -75,6 +75,12 @@
     @Query("SELECT g FROM User u JOIN u.groups g WHERE u = ?1 AND g.type != org.apromore.dao.model.Group.Type.GROUP")
     Set<Group> findCompulsoryGroups(User user);
 
+    /**
+     * @return all groups
+     */
+    @Query("SELECT g FROM Group g")
```
Comment (nolantellis) [inline Apromore-Database/src/main/java/org/apromore/dao/GroupRepository.java:81]: You can Remove this method and USe findAll()

**Review by nolantellis — CHANGES_REQUESTED** [review]

**Review by franklma — APPROVED** [review]
